workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/WorkflowTree.kt

```diff
@@ -0,0 +1,85 @@
+package com.squareup.workflow1.traceviewer
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
+/**
+ * Since the logic of Workflow is hierarchical (where each workflow may have parent workflows and/or
+ * children workflows, a tree structure is most appropriate for representing the data rather than
+ * using flat data structures like an array.
+ *
+ * TBD what more metadata should be involved with each node, e.g. (props, states, # of render passes)
+ */
+public data class WorkflowNode(
+  val id: String,
+  val name: String,
+  val children: List<WorkflowNode>
+)
+
+/**
+ * Since the workflow nodes present a tree structure, we utilize a recursive function to draw the tree.
+ * The Column holds a subtree of nodes, and the Row holds all the children of the current node.
+ */
+@Composable
+public fun DrawWorkflowTree(
+  node: WorkflowNode,
+  modifier: Modifier = Modifier,
+) {
+  Column(
```

Comment:
This works fine for now but eventually, we're probably gonna have to get _lazy_ with it. I'm not familiar with it but I recently learned that Compose has a `LazyLayout` Composable that, similar to `LazyRow` and `LazyColumn`, lets you render a bunch of stuff but is smart enough to only render things that are actually in view. Since these traces are gonna get big, we'll start to see lagging when scrolling. We might already with your `workflow-300` trace!

I can probably bully one of the UI System folks for more info on how we could potentially use that here, since the documentation is really sparse and `LazyLayout` is still experimental.